clang/include/clang/Analysis/Analyses/LifetimePset.h

```diff
@@ -47,8 +47,17 @@ struct Variable : public ContractVariable {
     return Variable(ContractVariable::returnVal(), nullptr);
   }
 
-  // TODO: is this what we want when derefs are involved?
-  bool isBaseEqual(const Variable &O) const { return Var == O.Var; }
+  bool isParent(const Variable &O) const {
+    auto isPrefixOf =
+        [this](
+            const llvm::SmallVector<const FieldDecl *, 4> &OtherFDs) -> bool {
+      if (OtherFDs.size() < FDs.size())
```

Comment:
I wonder if we want to have less or equal here to make psets invalid in invalidateVar() that contain `O` without extra fields and derefs?

Reply:
With less equal some of the existing test cases fail which seems wrong to me.
For example the dereferencing a danglin pointer warning is no longer produced:
void ref_leaves_scope() {
  int *p;
  {
    int i = 0;
    p = &i;
    *p = 2; // OK
  }         // expected-note {{pointee 'i' left the scope here}}
  *p = 1;   // expected-warning {{dereferencing a dangling pointer}}
}